apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -806,55 +806,67 @@ func (m *ModelManagerAPI) ListModelSummaries(ctx context.Context, req params.Mod
 		return result, errors.Trace(err)
 	}
 
-	modelInfos, err := m.state.ModelSummariesForUser(userTag, req.All && m.isAdmin)
+	var modelInfos []model.ModelSummary
+	if req.All {
+		if !m.isAdmin {
+			return result, fmt.Errorf("%w: cannot list all models as non-admin user", apiservererrors.ErrPerm)
+		}
+		modelInfos, err = m.modelService.ListAllModelSummaries(ctx)
+	} else {
+		modelInfos, err = m.modelService.ListModelSummariesForUser(ctx, userTag.Id())
+	}
 	if err != nil {
 		return result, errors.Trace(err)
 	}
+
 	err = m.fillInStatusBasedOnCloudCredentialValidity(ctx, modelInfos)
 	if err != nil {
-		return result, errors.Trace(err)
+		return result, apiservererrors.ServerError(
+			errors.Annotatef(err, "listing model summaries: filling in status for missing cloud credential"),
+		)
 	}
 
 	for _, mi := range modelInfos {
-		lastConnection, err := m.accessService.LastModelLogin(ctx, userTag.Id(), coremodel.UUID(mi.UUID))
-		if errors.Is(err, accesserrors.UserNeverAccessedModel) {
-			mi.UserLastConnection = nil
-		} else if errors.Is(err, modelerrors.NotFound) {
-			// TODO (aflynn): Once models are fully in domain, replace the line
-			// below with a `continue`. When models are still in state, this
-			// case is triggered because the model cannot be found in the domain
-			// db. Generally, it should only be triggered if the model has been
-			// removed since we got the UUID.
-			mi.UserLastConnection = nil
-		} else if err != nil {
-			return result, errors.Annotatef(err, "getting model last login time for user %q on model %q", userTag.Name(), mi.Name)
+		summary, err := m.makeModelSummary(mi)
+		if err != nil {
+			result.Results = append(result.Results, params.ModelSummaryResult{Error: apiservererrors.ServerError(err)})
 		} else {
-			mi.UserLastConnection = &lastConnection
+			result.Results = append(result.Results, params.ModelSummaryResult{Result: summary})
 		}
-		summary := m.makeModelSummary(mi)
-		result.Results = append(result.Results, params.ModelSummaryResult{Result: summary})
 	}
 	return result, nil
 
 }
 
-func (m *ModelManagerAPI) makeModelSummary(mi state.ModelSummary) *params.ModelSummary {
+func (m *ModelManagerAPI) makeModelSummary(mi model.ModelSummary) (*params.ModelSummary, error) {
+	credTag, err := mi.CloudCredentialKey.Tag()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if !names.IsValidCloud(mi.CloudName) {
```

Comment:
I think we should remove this check and code. If the data has come from our database then the cloud name should be valid. It should be impossible for a cloud name that isn't valid to end up in storage. We should check that the logic in this call is being applied on the service layer when making new clouds and remove this func from names completely.

Reply:
This check is here because of the potential for a panic with `NewCloudTag`. I do agree that this check isn't necessary and I would remove it otherwise, but I think that the potential for a controller panic is too high a risk to pay not to do this check.

Reply:
I think the potential for having this business logic blow up is just as bad as the panic. Can I suggest we cut a new version of names that removes the panic and find any calls that we have to is valid for cloud over to our own validator inside of Juju.

Reply:
Thinking about this a little more we are also making an explicit decision that names are a facade only thing and are restricted to just that API. However our API is also the service it self where other components in Juju will be using it. i.e bootstrap worker. It does not make any sense to be relying on the names package for prescribing what is good taste and what isn't within Juju.

Reply:
Agreed, see my PR #17775 where I have already started doing this for the username

Reply:
In terms of doing this generally, I agree that we should have a non panic version of names, but still, if we are using names over the wire, we will have to construct the values to send at some point, and this will inevitably require checks like this. 

Imo, ideally we would have functions that convert the domain/core type to a names type without verifying the name (i.e. not returning an error and not panicking) on the assumption that if the type is wrapped in some domain type then it cannot be an invalid type, since it has come directly from the database. We shouldn't be using the domain types to manually construct name types without checking for errors.

Reply:
Leave the names package. Bring in the logic from the names package into the service. We can strangle out the names package at a further date.